doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1100 @@
+             +--ro ALARM_LIST* [id]
+                +--ro id              uint64
+                +--ro resource?       string
+                +--ro text?           string
+                +--ro time-created?   yang:date-and-time
+                +--ro type-id?        string
+                +--ro severity?       event:severity-type
+                +--ro acknowledged?   boolean
+```
+
+openconfig alarms yang is defined at [here](https://github.com/openconfig/public/blob/master/release/models/system/openconfig-alarms.yang)
+
+### 3.3.2 CLI
+#### 3.3.2.1 Exec Commands
+```
+sonic# alarm acknowledge <seq-id-of-raised-alarm>
+```
+An operator can acknolwedge a raised alarm. This indicates that the operator is aware of the fault condition and considers the condition not catastrophic. 
+Acknowledging an alarm updates system health parameter and thereby system LED by removing the particular alarm from status consideration.
+
+The alarm record in the ALARM table is marked with is_acknowledged field set to true.
+
+```
+sonic# event profile <profile-name>
+```
+The command takes name of specified file, validates it for its syntax and values; merges it with its internal static map of events *static_event_map*.
+
+#### 3.3.2.2 Configuration Commands
+```
+sonic(config)# logging server <ip> [log|event] 
+```
+Note: The 'logging server' command is an existing, already supported command. 
+It is only enhanced to take either 'log' or 'event' to indicate either native syslog messages or syslog messages corresponding to events alone are sent to the remote host.
+Support with VRF/source-interface and configuring remote-port are all backward comaptible and will be applicable to either 'log' or 'event' options.
+
+#### 3.3.2.3 Show Commands
+```
+sonic# show system health
+System Health: Normal, Warning or Error
+
+sonic# show event profile
+Event Profile: my_ev_prof.json
+
+sonic# show event [ detail | summary | severity <sev> | timestamp from <from-ts> to <to-ts> | recent <5min|1hr|1day> | sequence-id from <from-seq> to <to-seq> ] 
+
+'show event' commands would display all the records in EVENT table.
+
+sonic# show event 
+Sq No   State        Name                             Timestamp                   Description
+-----   --------     ------------------------------   --------------------------- --------------
+2292    -            TAM_SWITCH_ID_CHANGE             Wed Feb 10 18:08:27 2021    processSwitchTblEvent: Received new switch-id as 3        
+2291    raised       TEMPERATURE_EXCEEDED             Wed Feb 10 18:08:24 2021    temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees
+182     cleared      TEMPERATURE_EXCEEDED             Wed Feb 10 09:52:12 2021    temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees
+134     raised       TEMPERATURE_EXCEEDED             Wed Feb 10 09:32:01 2021    temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees
+        
+sonic# show event details 
```

Comment:
Is there a limit number of the events? If there are thousands of them will they all be displayed in one shot?

Reply:
There is no limit on number of events displayed here. The command tries to display **ALL** the records in the table. However user can stop the output at any point of the time as it uses pagination.